### ISSUE TITLE
One Question regarding ElevatorCoordinator

### ISSUE THREAD

**Issue by ssemichael** [mobby1982/elevator-sim#1]
Hi,
 i am reading your elevator sim code,
 does the code below mean for when a request is created, you send it to all the elevators?
  case SendRequest =>
      context.children.foreach { e =>
        e ! FromFloor(generateRequest)
      }

**Comment by mobby1982** [comment]
You are correct, on receiving the SendRequest message, all the "children" of the current actor are sent the request FromFloor.

**Comment by ssemichael** [comment]
there is a particular reason for this logic? naturally one request only may one assigned to one elevator

**Comment by mobby1982** [comment]
No its a design choice i made.

**Comment by ssemichael** [comment]
ok. what leads to the design choice? or just for exercising. 
i am writing a elevator sim using akka as well. so i am interested in how to dispatch well

**Comment by mobby1982** [comment]
so if you look at the  CreateElevator message at the end the actor sends its self a message SendRequest. So the ElevatorCordinator is "pushing" the messages to each elevator in the system.  You could have it where each individual elevator "pulls" the message off from the elevator coridnator.  I made the choice of pushing the messages.

**Comment by ssemichael** [comment]
thanks for explaining.
so this version has no optimized dispatch logic to decide which elevator to assign a FromFloor

**Comment by ssemichael** [comment]
correct?

**Comment by mobby1982** [comment]
yes there is no dispatch logic, here when the elevator is done doing its quota of requests it just asks for more. now while thinking about dispatch logic there are a lot of ways to implement that.
one way would be to queue up floor requests and have each elevator fulfills only one request then at max you can only process requests == no of elevators.
another way would be to continuously check the requests and coordinate the elevators. 
ex: the elevator is at floor 0 and there is a request for floor 10 while trying to fulfill this request , incomes another request for floor 6 in the same direction, fulfill it too.
